util/eventbus/bus.go

```diff
@@ -80,18 +86,18 @@ func (b *Bus) Close() {
 	b.router.StopAndWait()
 
 	var clients set.Set[*Client]
-	b.topicsMu.Lock()
+	b.clientsMu.Lock()
```

Comment:
(pre-existing, but overlooked in a previous pass) I'd probably make this `lock / defer unlock` rather than capturing. If we're stopping we presumably do not want to allow other goroutines to continue updating the state of the object while we're closing the existing clients.